atomics/include/desul/atomics/GCC.hpp

```diff
@@ -89,6 +89,43 @@ DESUL_GCC_INTEGRAL_OP_ATOMICS(MemoryOrderRelaxed, MemoryScopeCore)
 DESUL_GCC_INTEGRAL_OP_ATOMICS(MemoryOrderSeqCst, MemoryScopeNode)
 DESUL_GCC_INTEGRAL_OP_ATOMICS(MemoryOrderSeqCst, MemoryScopeDevice)
 DESUL_GCC_INTEGRAL_OP_ATOMICS(MemoryOrderSeqCst, MemoryScopeCore)
+
+template <typename T, class MemoryOrder, class MemoryScope>
+std::enable_if_t<!std::is_trivially_copyable<T>::value, T>
```

Comment:
Why did you define these here rather than in `Compare_Exchange_GCC.hpp `

Reply:
because it actually requires the lock based stuff, and is not the fundamental compare exchange functionality which underpins all the other fallbacks.

Reply:
I guess we could forward declare the locking mechanisms ... but I think its nice that that file has only the primitives.